### ISSUE TITLE
Rewrite query ignores Table alias

### ISSUE THREAD

**Issue by adarshkeshan** [adamchainz/django-mysql#464]
Summary:
Rewriting query for index suggestions, ignore table alias causing the generated query to augment the actual query at the wrong place, which causes it to generate invalid queries.

`SELECT TABLE_ALIAS.FIELD1 FROM TABLE_NAME TABLE_ALIAS WHERE ( TABLE_ALIAS.FIELD2=Value AND (/*QueryRewrite':index=`TABLE_NAME` IGNORE `INDEX_TO_IGNORE`*/1))`

Gets rewritten to:

`SELECT TABLE_ALIAS.FIELD1 FROM TABLE_NAME IGNORE INDEX (INDEX_TO_IGNORE) TABLE_ALIAS WHERE ( TABLE_ALIAS.FIELD2=Value AND (1))`

Rather than

`SELECT TABLE_ALIAS.FIELD1 FROM TABLE_NAME TABLE_ALIAS IGNORE INDEX (INDEX_TO_IGNORE) WHERE ( TABLE_ALIAS.FIELD2=Value AND (1))`

Basically the regex that rewrite uses ignore any table alias. Currently if I pass a querySet to another querySet, django creates an Alias for the table, which breaks the rewrite translation.

The bug lies in the regex `table_spec_re_template` at https://github.com/adamchainz/django-mysql/blob/master/django_mysql/rewrite_query.py#L169.  Which the rewrite uses to describe the table definition.

Django Version: 2.0.5

Database and version used: MySQL 5.7

Version: *e.g. Django-MySQL 1.0.1*

**Comment by adamchainz** [comment]
can you provide a failing test? Also put your django-mysql version just to check?

**Comment by adamchainz** [comment]
Thanks. I've made #473 to look at this but it's hard to make it fully generic.

**Comment by adamchainz** [comment]
I've abandoned that PR due to its difficulty, so closing the issue as well. If you need specific query hints you can always rewrite Django's SQL with [custom database instrumentation](https://docs.djangoproject.com/en/3.0/topics/db/instrumentation/).
